specs/dynamicdependencies/sample-2.md

```diff
@@ -0,0 +1,85 @@
+# Sample 2 - Fabrikam app using Contoso's Muffins package with smart class helpers
+
+Variant of Sample 1 using RAII classes to manage packageDependencyId, packageDependencyContext and DLL lifetimes.
+
+```MddPinPackageDependency``` will fail if there are no packages registered for the calling user satisfying the PackageDependency.
+
+## Win32
+
+```c++
+#include <MsixDynamicDependency.hpp>
+#include <wil/resource.h>
+
+class MddPackageDependencyContextAndDll
+{
+public:
+    MddPackageDependencyContextAndDll() = default;
+
+    ~MddPackageDependencyContextAndDll()
+    {
+        (void) LOG_IF_FAILED(MddRemovePackageDependency(m_packageDependencyContext));
+        (void) LOG_IF_WIN32_BOOL_FALSE(FreeLibrary(m_module.get()));
+    }
+
+    HRESULT Add(
+        _In_ PCWSTR packageDependencyId,
+        INT32 rank,
+        MddAddPackageDependency flags,
+        _In_ PCWSTR dllFilename)
+    {
+        RETURN_IF_FAILED(MddAddPackageDependency(
+            packageDependencyId, rank, flags, &m_packageDependencyContext , &m_packageFullName));
+        m_module = ::LoadLibrary(L"Contoso-Muffins");
+        RETURN_LAST_ERROR_IF(!m_module);
+        return S_OK
+    }
+
+    PCWSTR GetPackageFullName() const
+    {
+        Return m_packageFullName.get();
+    }
+
+    HMODULE GetModule()
```

Comment:
Constify

Reply:
OK but the return value needs to be non-const as Windows APIs taking HMODULE won't accept const HMODULE.

Make the method const returning non-const HMODULE? Should the attribute be mutable or GetModule use const_cast?

Or keep the method as non-const and can only be used by if folks have a non-const instance?